bokehjs/src/lib/models/renderers/glyph_renderer.ts

```diff
@@ -298,6 +304,11 @@ export class GlyphRendererView extends DataRendererView {
     const glsupport = this.has_webgl
 
     this.glyph.map_data()
+    this.decimated_glyph.map_data()
+    this.selection_glyph.map_data()
+    this.nonselection_glyph.map_data()
+    this.hover_glyph?.map_data()
+    this.muted_glyph.map_data()
```

Comment:
Can you speak about the implications of the current changes here and/or eventual plans? Naively, I would expect this to explode data size and computational overhead. Avoiding that duplication was the topic of https://github.com/bokeh/bokeh/issues/10403 It's possible to current PR accomplishes that already, and I am just missing it at a quick look, but any pointers would be appreciated.

Reply:
Only properties that differ between a derived glyph (e.g. hover glyph) and the base will be computed (`set_data()`) and mapped (`map_data()`). Thus, for all existing setups, the only additional cost is iteration over properties and definition of getters that are used to fetch data from the parent glyph. So far I didn't notice any performance implications, which can be verified by observing the run time of bokehjs' integration tests. One thing I to still have address is custom definitions in `_set_data()` and `_map_data()`, which is the primary reason for the PR being WIP.

Reply:
> Avoiding that duplication was the topic of https://github.com/bokeh/bokeh/issues/10403

I overlooked that issue and it seems that I covered quite a few of the points there though not all.